lib/wordcount.rb

```diff
@@ -0,0 +1,38 @@
+def shred line #split lines into words
+  line.split /[\?\.,_:;@#\n\$\^\*\<\>\%\&!{}\+\n\\"\)\(\-— ]+/
+  #personalize regex for 'word' definition
+end
+
+def add_to_list array_of_words, list #adds only new words to hash list
+  array_of_words.each do |word|
+    key = word.upcase.capitalize #downcase to prevent case mattering
+    if list.has_key? key #if word already exists, inc count
+      list["#{key}"]=list["#{key}"]+1
```

Comment:
Since `key` is a string, there's no need to interpolate it into a new string. `"#{key}"` is better written as `key`.

When you want to increment a value in Ruby, the typical syntax is: `foo += 1`.

Together: `list[key] += 1`